src/core_scan/Core_scan.ml

```diff
@@ -891,16 +893,50 @@ let scan ?match_hook config ((rules, invalid_rules), rules_parse_time) :
         Core_profiling.Debug { skipped_targets; profiling }
     | (Core_profiling.Time _ | Core_profiling.No_info) as x -> x
   in
-
   { res with matches; errors; extra }
 
+(*****************************************************************************)
+(* Pre and Post Processors Hook For Semgrep Pro / Extensions        *)
+(*****************************************************************************)
+
+module type Pre_and_post_processor = sig
+  type state
+
+  val pre_process : Rule.t list -> Rule.t list * state
+  val post_process : state -> Core_result.t -> Core_result.t
+end
+
+(* The default processor is the identity processor which does nothing. *)
+module No_Op_Processor : Pre_and_post_processor = struct
+  type state = unit
+
+  let pre_process rules = (rules, ())
+  let post_process () results = results
+end
+
+let hook_pre_and_post_processor =
+  ref (module No_Op_Processor : Pre_and_post_processor)
+
+(* Written with semgrep_with_rules abstracted to allow reuse across
```

Comment:
it's not called semgrep_with_rules anymore, it's called just  'scan()'.
so maybe rename sg_with_rules below as 'core_scan_func'